### PR TITLE
Bugfix/chirp sampling ISSUE 546

### PR DESCRIPTION
**Pull request by MatthewAitken**
Thank you for contributing to the IPFX, your work and time will help to
advance open science! 

# Overview:
This PR resolves Issue #546  - sweep extraction failing due to sampling rate for 12.5khz chirp samples, and fixes how sweep features are extracted

**Stimulus Ontology needs to be updated to include sweep types with names Chirp, ChirpA, ChirpB, ChirpC, and ChirpD**

# Addresses:
Addresses issue #546

# Type of Fix:

- [ ] Bug fix (non-breaking change which fixes an issue)
- [ ] New feature (non-breaking change which adds functionality)
- [x] Breaking change (fix or feature that would cause existing
      functionality to not work as expected)
- [ ] Documentation Change

# Solution:
During my investigation, I realized that the function `extract_sweep_features` in data_set_features.py was not doing what it was intended to be doing. In short, the variable `stimulus_name` never matched the keys in DETECTION_PARAMETERS, so the default detection parameters were always being used for `extract_sweep_features()`. 
As an example, `stimulus_name` would be `"Short Square"` for a short square stimulus, but the key in DETECTION_PARAMETERS was `"short_square"`

(`extract_cell_long_square_features()`, `extract_cell_short_square_features()`, and `extract_cell_ramp_features()` were working as intended.)

The solution to both of these issues was to update DETECTION_PARAMETERS - correcting the key issue for short square sweeps, and adding a new one for Chirp sweeps, which will set the filter_frequency to None.

# Changes:
* Updated how stimulus names are mapped to stimulus types (e.g. 
short_square = {
        "Short Square",
        "Short Square Threshold",
        "Short Square - Hold -60mV",
        "Short Square - Hold -70mV",
        "Short Square - Hold -80mV",
}
* Added Chirp stimulus type
* Added detection parameters for Chirp stimulus

# Validation:
Validated that unit tests pass with expected changes to Short Square Sweep Features (since they now use the correct Detection Parameters)
Validated the data in issue #546 (an nwb file with chirp sweeps) successfully passes run_pipeline

### Configuration details:

# Checklist
- [ ] My code follows
      [Allen Institute Contribution Guidelines](CONTRIBUTING.md)
- [ ] My code is unit tested and does not decrease test coverage
- [ ] I have performed a self review of my own code
- [ ] My code is well-documented, and the docstrings conform to
      [Numpy Standards](https://numpydoc.readthedocs.io/en/latest/format.html)
- [ ] I have updated the documentation of the repository where
      appropriate
- [ ] The header on my commit includes the issue number
- [ ] My code passes all tests
- [ ] I have updated the CHANGELOG.md with the description of changes understandable by end users

# Notes:
Use this section to add anything you think worth mentioning to the
reader of the issue

### CODE REVIEW

**Review by gouwens — COMMENTED** [review]
I think the changes make sense, but I would categorize them as breaking changes for accessing the sweep names via the ontology object (i.e., capitalizing the constant names) and for moving the definitions out of EphysDataSet into StimulusType. That's kind of evidenced by having to make the changes in run_feature_collection.py and run_feature_vector_extraction.py - those are in a sense examples of "user code", even though they are included with the package.

So I think as long as that is made clear, the PR seems good.

**Comment by gouwens** [comment]
Just to follow up - another option could be to keep both cases and have the definitions in both places for a time and include a deprecation warning.

File: ipfx/data_set_features.py [hunk 1]
```diff
@@ -102,6 +114,7 @@ def extractors_for_sweeps(sweep_set,
                           dv_cutoff=20., thresh_frac=0.05,
                           reject_at_stim_start_interval=0,
                           min_peak=-30,
+                          filter_frequency=10.,
```
Comment (njmei) [inline ipfx/data_set_features.py:117]: Why was this line added? Does this change the behavior for previously passing processing?

**Review by tmchartrand — COMMENTED** [review]
Good catch on the detection parameters issue! 
Regarding this as a fix for the chirp issue, I'm not quite sure we actually want to always set no filtering as a default for chirps. @ru57y34nn do you know if all chirp sweeps are sampled at lower rate? Some of the external data I've used may be different too, I can try to check. 

My existing fix for this was to add a check to the filtering function and ignore invalid filtering parameters with a warning (still maybe a good idea for robustness). I'll let others chime in on whether they think fully disabling filters for chirps is a good idea.

File: ipfx/stimulus.py [hunk 2]
```diff
@@ -45,29 +97,10 @@ def __init__(self, stim_ontology_tags=None):
...
-        self.search_names = ("Search",)
-        self.test_names = ("Test",)
-        self.blowout_names = ( 'EXTPBLWOUT', )
-        self.bath_names = ( 'EXTPINBATH', )
-        self.seal_names = ( 'EXTPCllATT', )
-        self.breakin_names = ( 'EXTPBREAKN', )
-        self.extp_names = ( 'EXTP', )
+        for stimulus_type, names in _STIMULUS_TYPE_NAME_MAPPING.items():
+            # Sets helper attributes like StimulusOntology.RAMP_NAMES
+            setattr(self, f"{stimulus_type.upper()}_NAMES", names)
```
[the remark this answers is not in the document]
Comment (MatthewAitken) [reply ipfx/stimulus.py]: reverted to more explicit definition

**Review by tmchartrand — APPROVED** [review]
looks good to me, thanks.

**Review by rpmcginty — APPROVED** [review]
lgtm, minor comments

**Review by njmei — APPROVED** [review]

File: ipfx/stimulus.py [hunk 3]
```diff
@@ -4,6 +4,77 @@
 
 import allensdk.core.json_utilities as ju
 
+from enum import Enum
+
+
+_STIMULUS_TYPE_NAME_MAPPING = {
+    # Maps stimulus type to set of names
+    "ramp": {"Ramp"},
+    "long_square": {
```
[the remark this answers is not in the document]
Comment (MatthewAitken) [reply ipfx/stimulus.py]: I had it originally set to the keys, but was requested to explicitly list them out. Enum members (StimulusType.Ramp, StimulusType("ramp"), etc) have always been hashable.
